### ISSUE TITLE
Delay after executing `fig run` script

### ISSUE THREAD

**Issue by vandrieu** [withfig/fig#2149]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

macOS 13.0.0 (22A380)

### Expected behaviour

Running a command through `fig run` should not hang for 2 seconds

### Actual behaviour

If I type `ls` in my terminal, then the list of files is printed and I can type my next command immediately.

But If I run `fig run test-ls` (`test-ls` is a fig script which runs `ls`), then the list of files is printed immediately but the process doesn't exit immediately. It hangs for 2 seconds. Only then I can type my next command.

### Steps to reproduce

- Create a fig script called `test-ls` that runs the command `ls`
- Run `time ls` to get the execution time for `ls` (I get 4 to 5 milliseconds)
- Run `time fig run test-ls` (I get 2 to 3 seconds)

Note: same issue happens with "fig access" (`fig ssh`).

### Environment

```yaml
fig-details:
  - 2.10.1
hardware-info:
  - model: 
  - model-id: 
  - chip-id: Apple M1 Pro
  - cores: 10
  - mem: 32.00 GB
os-info:
  - macOS 13.0.0 (22A380)
environment:
  - shell: /bin/zsh
  - terminal: terminal
  - cwd: /Users/vincent
  - exe-path: /opt/homebrew/bin/fig
  - install-method: brew
  - env-vars:
    - FIGTERM_SESSION_ID: 134437c2-59cd-42f1-8e81-383512ebef65
    - FIG_HOSTNAME: m1pro.home
    - FIG_SET_PARENT_CHECK: 1
    - FIG_TERM: 2.10.1
    - PATH: /Users/vincent/.npm-global/bin:/Users/vincent/scripts:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/vincent/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/Users/vincent/.local/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/Wireshark.app/Contents/MacOS:/usr/bin:/bin:/Users/vincent/.cache/zsh4humans/v5/fzf/bin:/Users/vincent/.fig/bin:/Users/vincent/Library/Python/3.9/bin
    - SHELL: /bin/zsh
    - TERM: xterm-256color
    - __CFBundleIdentifier: com.apple.Terminal
    - FIG_PID: 26235
    - FIG_SET_PARENT: 134437c2-59cd-42f1-8e81-383512ebef65
```

**Comment by grant0417** [comment]
This will be resolved in 2.11

**Comment by grant0417** [comment]
2.11 is live, let me know if that fixes the issue
